modules/compute/src/integrationTest/java/org/apache/ignite/internal/compute/threading/ItComputeApiThreadingTest.java

```diff
@@ -141,13 +150,60 @@ private static Set<ClusterNode> justNonEntryNode() {
         return Set.of(unwrapIgniteImpl(CLUSTER.node(1)).node());
     }
 
+    @CartesianTest
+    void taskExecutionFuturesCompleteInContinuationsPool(
+            @Enum ComputeMapReduceOperation mapReduceOperation,
+            @Enum TaskExecutionAsyncOperation executionOperation
+    ) {
+        TaskExecution<?> execution = mapReduceOperation.executeOn(computeForPublicUse());
+
+        CompletableFuture<Thread> completerFuture = executionOperation.executeOn(execution)
+                .thenApply(unused -> currentThread());
+
+        assertThat(completerFuture, willBe(
+                either(is(currentThread())).or(asyncContinuationPool())
```

Comment:
I don't think we need `is` if there's already a `be`. `a(currentThread())` maybe? I doubt it exists

Reply:
`either()` requires a `Matcher` to be passed. I wanted to avoid `is` here, but alas.
I did not find `a()`. There is `isA()`, but it's a sinonym for `instanceOf()`.